subsys/bootloader/bl_validation/bl_validation.c

```diff
@@ -140,18 +140,20 @@ validation_info_find(u32_t start_address, u32_t search_distance)
 
 
 #ifdef CONFIG_SB_VALIDATE_FW_SIGNATURE
-static bool validate_signature(u32_t fw_src_address,
-				const struct fw_info *fwinfo,
+static bool validate_signature(const u32_t fw_src_address,
+				const u32_t fw_size,
 				const struct fw_validation_info *fw_val_info,
 				bool external)
 {
-	int retval = bl_crypto_init();
+	bool validated = false;
+	int init_retval = bl_crypto_init();
 
-	if (retval) {
-		PRINT("bl_crypto_init() returned %d.\n\r", retval);
+	if (init_retval) {
+		PRINT("bl_crypto_init() returned %d.\n\r", init_retval);
 		return false;
 	}
 
+	int retval = -EINVAL;
```

Comment:
nit: Should this be moved down some lines so that its closer to where it is used?

Reply:
Alternatively, what if we moved this declaration to the line where we assign it to the return value of `rot_verify`.

Then we remove the `if (retval != 0)` check outside that scope and print that error inside that scope instead.

The check for `if (validated)` should suffice.

THis to improve readability.

Reply:
And `return false` inside that scope instead of doing a `break`

Reply:
I refactored the for loop. Now there are more direct returns, and no breaks. I don't use the validated flag, since I return directly instead.